processor/src/main/java/org/mapstruct/ap/internal/model/CollectionAssignmentBuilder.java

```diff
@@ -0,0 +1,163 @@
+/**
+ *  Copyright 2012-2017 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
```

Comment:
That is too a possible way to go. I am fine with any approach, if it reduces the complexity of the class :)